### DISCUSSION TITLE
Strange triangulation(for all partitioning).

### DISCUSSION THREAD

**Discussion by Myszkoj** [recastnavigation/recastnavigation#562] [Q&A] [answered]
Hello,
I noticed weird artifacts in all triangulation types.
![recast bug](https://user-images.githubusercontent.com/9519504/148435759-3a6ce983-0059-4169-be2e-2c6211ffb911.png)
![image](https://user-images.githubusercontent.com/9519504/148438243-70577c95-4144-48c7-a4bb-639371259241.png)
![image](https://user-images.githubusercontent.com/9519504/148438343-bf4ab499-95a8-475e-a278-fc1976ca827f.png)
![image](https://user-images.githubusercontent.com/9519504/148439895-3130d13c-e1e4-43e9-a3fc-bbeeab8df76c.png)

My current configuration for rcConfig(values copy-pased from the editor):
```
tileSize               = 0;
borderSize             = 0;
cs                     = 0.1;  //<-- I have tried different resolutions, but it doesn't change much.
ch                     = 0.1;  //<--
walkableSlopeAngle     = 45Deg;
walkableHeight         = (int)ceilf(2.0 / m_cfg.ch);
walkableClimb          = (int)floorf(0.5 / m_cfg.ch);
walkableRadius         = 0; //<-- This sometimes helps, but I need mesh without erosion in my pathfinding.
minRegionArea          = 80;
mergeRegionArea        = 1024;
maxEdgeLen             = (int)(100.0 / m_cfg.cs);
maxSimplificationError = 1.3f; // <-- The only thing that I didn't tried yet.
maxVertsPerPoly        = 8;
detailSampleDist       = m_cfg.cs * 2.0; //<-- Changing this values doesn't help.
detailSampleMaxError   = m_cfg.ch * 3.0; //<--

// Calculated during build process:
m_cfg.width   = 0;
m_cfg.height  = 0; 
m_cfg.bmin[0] = 0.f; m_cfg.bmin[1] = 0.f; m_cfg.bmin[2] = 0.f;
m_cfg.bmax[0] = 0.f; m_cfg.bmax[1] = 0.f; m_cfg.bmax[2] = 0.f; 
```

I used this model, but it has to be rotated (doesn't work in RecastDemo):
https://www.sendspace.com/file/td582c

Any ideas what may help?

**Comment by memononen** [comment]
There are some cases where the detail mesh height calculation can fail when the agent radius is 0.
The problem is in RecastMeshDetail.cpp / getHeightData() where it tries to match the polygons back to the voxels.

**Comment by grahamboree** [comment, marked as answer]
If you wish to avoid eroding navmesh edges, you should skip calling `rcErodeWalkableArea` instead of setting the agent radius to 0 for the reasons Mikko outlines
